kie-server-parent/kie-server-controller/kie-server-controller-impl/src/test/java/org/kie/server/controller/impl/storage/FileBasedKieServerTemplateStorageTest.java

```diff
@@ -0,0 +1,279 @@
+        container.setServerInstanceId(template.getId());
+        container.setServerTemplateId(template.getId());
+        container.setResolvedReleasedId(containerSpec.getReleasedId());
+        container.setContainerName(containerSpec.getContainerName());
+        container.setContainerSpecId(containerSpec.getId());
+        container.setUrl("http://fake.server.net/kie-server");
+        container.setStatus(containerSpec.getStatus());
+
+        return template;
+    }
+    
+    /**
+     * Retrieves the first template from our static map of server templates,
+     * using an Iterator over the value in the map. Asserts that we did,
+     * in fact, retrieve a template.
+     * @return The first server template
+     */
+    private ServerTemplate getFirstTemplateFromMap() {
+        Iterator<ServerTemplate> iter = templateMap.values().iterator();
+        ServerTemplate testTemplate = iter.hasNext() ? iter.next() : null;
+        assertNotNull("Unable to find a test server template!",testTemplate);
+        return testTemplate;
+    }
+    
+    /**
+     * Retrieves a template from the static instance of FileBasedKieServerTemplateStorage,
+     * based on the id found in the passed in ServerTemplate. Asserts that we
+     * did retrieve a template, and that the retrieved template is equal to 
+     * the one that was used as a source for the id.
+     * @param template The source instance of a ServerTemplate
+     * @return The retrieved ServerTemplate instance
+     */
+    private ServerTemplate loadTemplateWithAssertEquals(ServerTemplate template) {
+        ServerTemplate loadedTemplate = storage.load(template.getId());
+        assertNotNull("Unable to load template from storage",loadedTemplate);
+        assertEquals("Loaded template is not the one asked for",template,loadedTemplate);
+        return loadedTemplate;
+    }
+    
+    @BeforeClass
+    public static void beforeClass() {
+        System.setProperty(FileBasedKieServerTemplateStorage.SERVER_TEMPLATE_FILE_NAME_PROP, 
+                TEST_SERVER_TEMPLATE_FILENAME);
+        specManageService = new SpecManagementServiceImpl();
+        runtimeManagementService = new RuntimeManagementServiceImpl();
+        kieServerInstanceManager = Mockito.mock(KieServerInstanceManager.class);
+        ((RuntimeManagementServiceImpl)runtimeManagementService).setKieServerInstanceManager(kieServerInstanceManager);
+
+        templateMap = Maps.newConcurrentMap();
+        for (int x = 0; x < 3; x++) {
+            StringBuilder templateName = new StringBuilder("test server : ").append(x);
+            ServerTemplate template = createServerTemplateWithContainer(templateName.toString(),x+1);
+            ServerTemplateKey key = new ServerTemplateKey(template.getId(), template.getName());
+            templateMap.put(key, template);
+        }
+    }
+    
+    @Before
+    public void setup() {
+        storage = new FileBasedKieServerTemplateStorage();
```

Comment:
for tests (to avoid to set and reset system properties) we could have extra constructor that accepts file path directly.